src/VisualStudio/CSharp/Impl/Snippets/SnippetExpansionClient.cs

```diff
@@ -81,6 +84,12 @@ public override int GetExpansionFunction(IXMLDOMNode xmlFunctionNode, string bst
                 case "GenerateSwitchCases":
                     pFunc = new SnippetFunctionGenerateSwitchCases(this, SubjectBuffer, bstrFieldName, param);
                     return VSConstants.S_OK;
+                case "ArgumentValue":
+                    // For the internal ArgumentValue function, the snippet field name is expected to match the
+                    // parameter name, and the string passed to the function is a serialized SymbolKey allowing the
+                    // snippet function to resolve the original IParameterSymbol.
```

Comment:
do we generate hte symbolkey text manually?  or do we use the SymbolKey type itself to generate this?  Just making sure we don't take an implicit dependency on hte format of SymbolKeys (which are allowed to change from version to version).

Reply:
Generated by `SymbolKey`, and should also be documented at the location where the generated key is created.